src/main/java/org/jenkinsci/plugins/matrixauth/AuthorizationPropertyDescriptor.java

```diff
@@ -48,8 +50,34 @@ public interface AuthorizationPropertyDescriptor<T extends AuthorizationProperty
 
     T create();
 
+    default T createNewInstance(StaplerRequest2 req, JSONObject formData, boolean hasOptionalWrap)
```

Comment:
Why? This class is `@Restricted(NoExternalUse.class)`.

Reply:
AFAICT at this point `azure-ad` only reuses inheritance types from `matrix-auth`, IIRC since the last time I broke something there by changing `@Restricted` code here.

@timja Could you confirm?